keras_core/backend/torch/image.py

```diff
@@ -69,3 +72,149 @@ def resize(
         elif len(image.shape) == 3:
             resized = resized.permute((1, 2, 0))
     return resized
+
+
+AFFINE_METHODS = (
+    "nearest",
+    "bilinear",
+)
+AFFINE_FILL_MODES = {
+    "constant": "zeros",
+    "nearest": "border",
+    # "wrap",  not supported by torch
+    # "mirror",  not supported by torch
+    "reflect": "reflection",
+}
+
+
+def _apply_grid_transform(
+    img,
+    grid,
+    method="bilinear",
+    fill_mode="zeros",
+    fill_value=None,
+):
+    """
+    Modified from https://github.com/pytorch/vision/blob/main/torchvision/transforms/v2/functional/_geometry.py
+    """  # noqa: E501
+
+    # We are using context knowledge that grid should have float dtype
+    fp = img.dtype == grid.dtype
+    float_img = img if fp else img.to(grid.dtype)
+
+    shape = float_img.shape
+    # Append a dummy mask for customized fill colors, should be faster than
+    # grid_sample() twice
+    if fill_value is not None:
+        mask = torch.ones(
+            (shape[0], 1, shape[2], shape[3]),
+            dtype=float_img.dtype,
+            device=float_img.device,
+        )
+        float_img = torch.cat((float_img, mask), dim=1)
+
+    float_img = tnn.grid_sample(
+        float_img,
+        grid,
+        mode=method,
+        padding_mode=fill_mode,
+        align_corners=False,
+    )
+    # Fill with required color
+    if fill_value is not None:
+        float_img, mask = torch.tensor_split(float_img, indices=(-1,), dim=-3)
+        mask = mask.expand_as(float_img)
+        fill_list = (
+            fill_value
+            if isinstance(fill_value, (tuple, list))
+            else [float(fill_value)]
+        )
+        fill_img = torch.tensor(
+            fill_list, dtype=float_img.dtype, device=float_img.device
+        ).view(1, -1, 1, 1)
+        if method == "nearest":
+            bool_mask = mask < 0.5
+            float_img[bool_mask] = fill_img.expand_as(float_img)[bool_mask]
+        else:  # 'bilinear'
+            # The following is mathematically equivalent to:
+            # img * mask + (1.0 - mask) * fill =
+            # img * mask - fill * mask + fill =
+            # mask * (img - fill) + fill
+            float_img = float_img.sub_(fill_img).mul_(mask).add_(fill_img)
+
+    img = float_img.round_().to(img.dtype) if not fp else float_img
+    return img
+
+
+def affine(
```

Comment:
Are you able to add it for the NumPy backend as well? Fine to depend on scipy.

Reply:
Added.